cmd/k8sagent/unit/agent.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2012-2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package unit
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+	"github.com/juju/names/v4"
+	"github.com/juju/utils/voyeur"
+	"github.com/juju/worker/v2"
+	"github.com/juju/worker/v2/dependency"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/uniter"
+	jujucmd "github.com/juju/juju/cmd"
+	jujudagent "github.com/juju/juju/cmd/jujud/agent"
```

Comment:
We want to avoid bringing in this package due to bootstrap, and all that brings in.

Reply:
agreed, so just moved all agentconf related code in `jujud/agent` top level package to a separate package;